configurate-gson/src/main/java/ninja/leaping/configurate/gson/GsonConfigurationLoader.java

```diff
@@ -196,6 +196,9 @@ private void generateValue(JsonWriter generator, ConfigurationNode node) throws
             generateObject(generator, node);
         } else if (node.hasListChildren()) {
             generateArray(generator, node);
+        } else if (node.getKey() == null && node.getValue() == null) {
```

Comment:
I feel like there should be a hasValue() method on ConfigurationNode so that a potentially expensive getValue call may be avoided -- but that's out of the scope of this PR